src/main/java/duke/command/MarkCommand.java

```diff
@@ -0,0 +1,49 @@
+package duke.command;
+
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+/**
+ * Represents a command to mark or unmark a task as completed.
+ * Changes the completion status of the specified task.
+ */
+public class MarkCommand implements Command {
+    /** The 1-based index of the task to mark/unmark */
+    private final int index;
+    /** Whether to mark (true) or unmark (false) the task */
+    private final boolean mark;
```

Comment:
Is there a better way to name `mark` ?
```suggestion
    private final boolean isMarked;
```

Reply:
Thanks again!